flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -861,6 +861,14 @@ private void jobStatusChanged(
 		validateRunsInMainThread();
 
 		if (newJobStatus.isGloballyTerminalState()) {
+			scheduledExecutorService.execute(() -> {
+				for (Map.Entry<ResourceID, Tuple2<TaskManagerLocation, TaskExecutorGateway>> entry : registeredTaskManagers.entrySet()) {
+					Collection<ResultPartitionID> storedPartitions = partitionTable.stopTrackingPartitions(entry.getKey());
+					// if this call fails TaskExecutors will cleanup partitions regardless once we close the connections
+					entry.getValue().f1.releasePartitions(jobGraph.getJobID(), storedPartitions);
```

Comment:
This might be not needed then for cancel/suspend any more (Execution.sendReleaseIntermediateResultPartitionsRpcCall).